### ISSUE TITLE
[1.16.4] Potential Chisels & Bits issue

### ISSUE THREAD

**Issue by CptPICHU** [vectorwing/FarmersDelight#242]
**Description**
I used "Chisels & Bits 0.3.4" at 1.16.4. and "Farmers Delight 4.0" at 1.16.4.
There i encountered a heavy issue: Chiseled Blocks from C&B had mixed materials.
Oddly enough not all materials have been replaced. The Problem occured right after updating Farmer's Delight to 4.0.
As i did a downgrade to Farmer's Delight 0.3.2 with a world Backup the issues where gone. Materials where no longer mixed.
It is important to say that i use a 90+ Mods Modpack and i absolutely CANNOT say for sure, that Farmer's Delight is the main issue here. However, 0.4 caused the issue 0.3.2 did not. It MAY have something to do with the FD Forge Tag rewrites? I'm really not sure. I can provide you the tools to replicate the issue.

1. Download my EXACT Modpack "PICHUPack PRO" in the Technic Launcher.
2. Create a world and build some Chisels and Bits Blocks with optimally lots of mixed materials.
3. The Pack includes FD 0.3.2. Replace it with 0.4 and open the world again.
4. The issue should be visible by some Chiseled Blocks being different.

If you have ANY issues testing, please comment here! Also please read logs if you want. I can't find anything in there. (No, reallly i did not)
If the issue is caused by something else in my Pack, i'm sorry for that.
But i have my reasons to believe that there might be a conflict between FD and C&B.

**Steps to reproduce:**
Oh. Oops. See above. ^^'

**Expected behavior:**
Seemingly random Chisels & Bits materials should be mixed.

**Logs**
[latest.log](https://github.com/vectorwing/FarmersDelight/files/6208718/latest.log)

**Screenshots**
Well, chiseled Blocks are just made out of different materials after the issue.

**Mod list:**
MC/FORGE: 1.16.4 /  35.1.36
FARMERS DELIGHT: 0.4
CHISELS & BITS: 0.3.4
There are a lot. Didn't even know you could let your Minecraft die. Awesome - i guess.
HINT: In the List it's FD 0.3.2 BECAUSE the Modpack version is already debugged.
This is not relevant, because you actually need to update FD on existing world with C&B to reproduce.
[crash-2021-03-26_02.16.19-client.txt](https://github.com/vectorwing/FarmersDelight/files/6208735/crash-2021-03-26_02.16.19-client.txt)

**Comment by vectorwing** [comment]
I managed to replicate the issue on a minimal instance:

1. FD 0.3.2 + C&B 0.3.4 + Pretty much any mod that adds blocks;
2. Set up some chiseled structures out of modded blocks;
3. Close the game, upgrade FD to 0.4;
4. Open the world.

Chiseled blocks become shuffled in seemingly random manners, with some even adopting parts of Shepherd's Pie's states (despite it being a block that cannot be chiseled). Seems to be a re-occurrence of an issue that C&B fixed on 0.3.0. The blocks below are from Extended Caves and Create:

![image](https://user-images.githubusercontent.com/5566692/112574773-c356f480-8dcd-11eb-8328-da57bae5461b.png)

Honestly, I have no idea what's happening. Nothing unique in any logs at all, and it only happens when FD is upgraded. My only theory is that Forge shuffled a bunch of ID's when adding Shepherd's Pie into the world from version 0.4. Makes me wonder if I messed up something on register...

Maybe the C&B devs will be able to tell what's happening better, and whether I'll have to fix something on my end.
